RFC 58: Add a new test property to control displaying test state updates (#58)
web-platform-tests/rfcs: rfcs/disable_notify_test_state.md

> # RFC ??: Add a new test property to not displaying test state

Reviewer: ```suggestion
# RFC 58: Add a new test property to control displaying test states
```

## Summary

> Add a new test property called `notify_test_state` to not displaying

Reviewer: ```suggestion
Add a new test property called `notify_test_state` to control displaying
```

> test state when tests are running.

Reviewer: ```suggestion
test state updates while tests are running.
```

> test state when tests are running.
> 

Reviewer: ```suggestion
test state when tests are running.

This property will default to `true`, and can be set to `false` via the `setup()` function.
```

> ## Details

Reviewer: How does this interact with the `test_done_callbacks`? I believe those too cause `WindowTestEnvironment` to update the DOM, right?

Author: I don't think this impacts the `test_done_callbacks`, this PR just prevents `notify_test_state` to be invoked. We invoke the `test_done_callbacks` in `notify_result` function which is not impacted by this PR.

Reviewer: Yes, what I'm asking is if they could affect the tests! E.g. if a test has 10 subtests, aren't the `test_done_callbacks` invoked for each individual subtest finishing, and then each of those 10 updates causes DOM changes?

Put another way, I suspect that blocking just `notify_test_state` may not be sufficient.
